docs/kbs_protocol.md

```diff
@@ -0,0 +1,276 @@
+(TODO). If KBS cannot support the API specified in this field, it will directly disconnect the 
+link and ignore the request.
+
+- `api-input`
+
+Input parameters for the service API. For example, for the Get Key service API, KBC need to 
+provide KBS with key ID and other information. For the input parameters required by each service 
+API, please refer to the [service API document ]() (TODO) of KBS. If the format of the input 
+parameters does not meet the requirements of he corresponding API, the parameter error 
+information of the API will be returned in the response information in step 4.
+
+## Challenge
+
+After KBS receives the request, if the token is found to be empty or expired, KBS will return an 
+attestation challenge to KBC. The payload format is as follows:
+
+```json
+{
+    "challenge": {
+	    /* To ensure the freshness of evidence, Base64 coding. */
+	    "nonce": "",
+	    /* Extra parameters to support some special HW-TEE attestation. */
+	    "extra-params": {}
+	}
+}
+```
+
+- `nonce`
+
+The fresh number passed to KBC. KBC needs to place it in the evidence sent to KBS in the next 
+step to prevent replay attacks.
+
+- `extra-params`
+
+The reserved extra parameter field which is used to pass the additional information provided by 
+the KBS when some specific HW-TEE needs to be attested.
+
+## Attestation
+
+After receiving the invitation challenge, KBC collects the attestation evidence of HW-TEE 
+platform and organizes it into the following payload format:
+
+```json
+{
+    "evidence": {
+	    /* The nonce included in challenge to prove the freshness of this message.
+	       Its hash needs to be included in HW-TEE evidence and signed by HW-TEE hardware. */
+	    "nonce": "",
+	    /* The public key generated by KBC in HW-TEE, it is valid until the next time an attestation is required.
+	       Its hash needs to be included in HW-TEE evidence and signed by HW-TEE hardware. */
+	    "tee-pubkey": {
+		    "algorithm": "",
+		    "pubkey-length": "",
+		    "pubkey": ""
+		},
+	    /* The content of evidence. Its format is specified by Attestation-Service. */
+	    "tee-evidence": {}
+	}
+}
+```
+
+- `nonce`
+
+Nonce contained in challenge. It is used to prove the freshness of this message and prevent 
+replay attacks. Its hash needs to be signed by HW-TEE hardware as part of the custom field of 
+HW-TEE evidence (commonly referred to as `report data`).
+
+- `tee-pubkey`
+
+After KBC receives the attestation challenge, a pair of temporary asymmetric keys are generated 
+in HW-TEE. The private key is stored in HW-TEE. The public key and its description information 
+are exported and placed in the `tee-pubkey` field and sent to KBS together with evidence. The 
+hash of the `tee pubkey` field needs to be included in the custom field of HW-TEE evidence and 
+signed by HW-TEE hardware. This public key is valid until the next time KBC receives this KBS's 
+attestation challenge.
```

Comment:
Yes, it is necessary to include nonce and pubkey in the attestion report. In this case, we will use `HASH(nonce || key)` as the user data of the report. If there are new important data to be included in the report in the future, it can be easily expanded as `HASH(nonce || key || other)`.